ehtim/survey.py

```diff
@@ -0,0 +1,635 @@
+            maxit (int): maximum number of iterations if image does not converge
+            stop (float): convergence criterion for imaging
+            fov (int): image field of view in uas
+            npixels (int): number of image pixels
+            reverse_taper_uas (int): fwhm of gassuain in uas to reverse taper observation
+            uv_zblcut (float): maximum uv-distance to which is considered short baseline flux
+            SEFD_error_budget (dict): SEFD percentage error for each station
+
+
+        Returns:
+
+            (dict): dict containing all non-varying survery parameters
+
+    """
+
+    # take all arguments and put them in a dict
+    args = list(locals().keys())
+    params_fixed = {}
+    for arg in args:
+        params_fixed[arg] = locals().get(arg)
+
+    return params_fixed
+
+def create_survey_psets(zbl=[0.6], sys_noise=[0.02], avg_time=['scan'], prior_fwhm=[40],
+                     sc_phase=[0], xdw_phase=[10], sc_ap=[0], xdw_ap=[1], amp=[0.2], cphase=[1], logcamp=[1],
+                     simple=[1], l1=[1], tv=[1], tv2=[1], flux=[1], epsilon_tv=[1e-10]):
+    """
+    Create a dataframe given all survey parameters. Default values will create an example dataframe but these values should be adjusted for each specific observation
+
+        Args:
+
+           zbl (array): compact flux value (Jy)
+           sys_noise (array): percent addition of systematic noise
+           avg_time (array): in seconds or 'scan' for scan averaging
+           prior_fwhm (array): fwhm of gaussian prior in uas
+           sc_phase (array): number of iterations to perform phase-only self-cal
+           xdw_phase (array): multiplicative factor for data weights after one round of phase-only self-cal
+           sc_ap (array): number of iterations to perform amp+phase self-cal
+           xdw_ap (array): multiplicative factor for data weights after one round of amp+phase self-cal
+           amp (array): data weight to be placed on amplitudes
+           cphase (array): data weight to be placed on closure phases
+           logcamp (array): data weight to be placed on log closure amplitudes
+           simple (array): regularizer weight for relative entropy, favoring similarity to prior image
+           l1 (array): regularizer weight for l1 norm, favoring image sparsity
+           tv (array): regularizer weight for total variation, favoring sharp edges
+           tv2 (array): regularizer weight for total squared variation, favoring smooth edges
+           flux (array): regularizer weight for total flux density, favoring final images with flux close to zbl
+           epsilon_tv (array): epsilon value used in definition of total variation - rarely need to change this
+
+        Returns:
+
+            (DataFrame): pandas DataFrame containing all combination of parameter sets along with index values
+
+    """
+
+    # take all arguments and put them in a dict
+    args = list(locals().keys())
```

Comment:
I'm a little horrified at `args = list(locals().keys())` but it gets the job done!